pkg/api/v1/schema.go

```diff
@@ -19,346 +19,474 @@
 
 package v1
 
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+)
+
 //go:generate swagger-doc
 
 /*
  ATTENTION: Rerun code generators when comments on structs or fields are modified.
 */
 
-// http://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html
-type CloudInitDataSourceNoCloud struct {
-	// Reference to a k8s secret that contains NoCloud userdata
-	UserDataSecretRef string `json:"userDataSecretRef,omitempty"`
-	// The NoCloud cloud-init userdata as a base64 encoded string
-	UserDataBase64 string `json:"userDataBase64"`
-	// The NoCloud cloud-init metadata as a base64 encoded string
-	MetaDataBase64 string `json:"metaDataBase64,omitempty"`
-}
-
-// Only one of the fields in the CloudInitSpec can be set
-type CloudInitSpec struct {
-	// Nocloud DataSource
-	NoCloudData *CloudInitDataSourceNoCloud `json:"nocloud"`
-
-	// Add future cloud init datasource structures below.
+// Represents a cloud-init nocloud user data source
+// More info: http://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html
+type CloudInitNoCloudSource struct {
+	// UserDataSecretRef references a k8s secret that contains NoCloud userdata
+	// + optional
+	UserDataSecretRef *v1.LocalObjectReference `json:"secretRef,omitempty"`
+	// UserDataBase64 contains NoCloud cloud-init userdata as a base64 encoded string
+	// + optional
+	UserDataBase64 string `json:"userDataBase64,omitempty"`
 }
 
 type DomainSpec struct {
-	Memory  Memory   `json:"memory"`
-	Type    string   `json:"type"`
-	OS      OS       `json:"os"`
-	SysInfo *SysInfo `json:"sysInfo,omitempty"`
-	Devices Devices  `json:"devices"`
-	Clock   *Clock   `json:"clock,omitempty"`
-}
-
-type Memory struct {
-	Value uint   `json:"value"`
-	Unit  string `json:"unit"`
+	// Resources describes the Compute Resources required by this vm.
+	Resources ResourceRequirements `json:"resources,omitempty"`
+	// Firmware
+	// +optional
+	Firmware *Firmware `json:"firmware,omitempty"`
+	// Clock sets the clock and timers of the vm.
+	// +optional
+	Clock *Clock `json:"clock,omitempty"`
+	// Features like acpi, apic, hyperv
+	// +optional
+	Features *Features `json:"features,omitempty"`
+	// Devices allows adding disks, network interfaces, ...
+	Devices Devices `json:"devices"`
+}
+
+type ResourceRequirements struct {
+	// Initial is a description of the initial vm resources.
+	// Valid resource keys are "memory" and "cpu".
+	// +optional
+	Initial v1.ResourceList `json:"initial,omitempty"`
```

Comment:
requests sounds good to me. There was also one a PR some discussion regarding  to this. Some people wanted to express here really what the VM is allowed to consume on the host (meaning guest ram + vga ram + qemu overhead). I think it is better to let people define the guest ram value, and let the system generate pods which take all these overheads into account (so that the scheduele can act accordingly). I also think that this is consistent from the intended usage of the resources section in k8s. No one would expect, that you have  to take the pause-infra containeer from k8s into account,  when you specify your resource requirements.

The overhead calculation could be static (like in oVirt), or pluggable.

Reply:
> requests sounds good to me.

alrighty, I'm on board with that. 

> I think it is better to let people define the guest ram value, and let the system generate pods which take all these overheads into account (so that the scheduele can act accordingly).

agreed

Reply:
Done